### PR TITLE
Convert typedef statements to using statements

### PR DESCRIPTION
**Pull request by timmah**
### Background

* Wish to convert typedef statements to using statements

### Purpose of Pull Request

* Convert typedef statements to using statements
* [Fixes Jayenne Redmine Issue #2204](https://rtt.lanl.gov/redmine/issues/2204)

### Description of changes

* Convert typedef statements to using statements via clang-tidy
* Adds `CMAKE_EXPORT_COMPILE_COMMANDS` to top-level `CMakeLists.txt` to assist clang-tidy

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [x] Travis/Appveyor CI checks pass
  * [x] Code coverage does not decrease
  * [x] [Valgrind test passes](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Toss3 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Trinitite checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,6 +15,9 @@ set(ddesc "An object-oriented component library supporting radiation")
 string(APPEND ddesc " transport applications.")
 project( Draco DESCRIPTION ${ddesc} VERSION 7.8 LANGUAGES CXX C)
 
+# Export compile commands for compiler tools
+set(CMAKE_EXPORT_COMPILE_COMMANDS TRUE)
```
Comment (attom) [inline CMakeLists.txt:19]: What's this change to the CMake? Otherwise, this PR LGTM

Comment (KineticTheory) [reply CMakeLists.txt:19]: @attom @timmah FYI - this is useful for LLVM tools with negligible side effects.

**Review by KineticTheory — APPROVED** [review]
I approve of these changes.  Let's see if the tests come back clean.

File: CMakeLists.txt (hunk 1 above)
Comment (timmah) [reply CMakeLists.txt:19]: This directs CMake to emit a "compilation database",  a JSON file with all the compile command lines listed. Clang tools like clang-tidy, clang-query etc use it to get things like compile arguments, include paths, etc.

**Comment by KineticTheory** [comment]
@timmah There are a couple of `clang-format` style checks that failed.  Can you patch those up?

**Pull request merged**